contracts/interchain-token-service/src/flow_limit.rs

```diff
@@ -38,26 +77,49 @@ pub fn set_flow_limit(
 }
 
 pub fn flow_out_amount(env: &Env, token_id: BytesN<32>) -> i128 {
-    let epoch = env.ledger().timestamp() / EPOCH_TIME;
     env.storage()
         .temporary()
-        .get(&DataKey::FlowOut(token_id, epoch))
+        .get(&DataKey::FlowOut(FlowKey {
+            token_id,
+            epoch: current_epoch(env),
+        }))
         .unwrap_or(0)
 }
 
 pub fn flow_in_amount(env: &Env, token_id: BytesN<32>) -> i128 {
-    let epoch = env.ledger().timestamp() / EPOCH_TIME;
     env.storage()
         .temporary()
-        .get(&DataKey::FlowIn(token_id, epoch))
+        .get(&DataKey::FlowIn(FlowKey {
+            token_id,
+            epoch: current_epoch(env),
+        }))
         .unwrap_or(0)
 }
 
-enum FlowDirection {
-    In,
-    Out,
+pub fn add_flow_in(
```

Comment:
How about moving `pub fn add_flow` within FlowDirection, and then you can call `FlowDirection::In::add_flow(...)` from ITS contract?